perun-oidc-server/src/main/java/cz/muni/ics/oauth2/web/OAuthConfirmationController.java

```diff
@@ -114,36 +155,82 @@ public String confirmAccess(Map<String, Object> model, Principal p) {
 			return HttpCodeView.VIEWNAME;
 		}
 
-		if (prompts.contains("none")) {
+		if (prompts.contains(NONE)) {
 			// if we've got a redirect URI then we'll send it
+			return sendRedirect(authRequest, model, client);
+		}
 
-			String url = redirectResolver.resolveRedirect(authRequest.getRedirectUri(), client);
+		model.put(AUTH_REQUEST, authRequest);
+		model.put(CLIENT, client);
 
-			try {
-				URIBuilder uriBuilder = new URIBuilder(url);
+		String redirect_uri = authRequest.getRedirectUri();
+		model.put(REDIRECT_URI, redirect_uri);
 
-				uriBuilder.addParameter("error", "interaction_required");
-				if (!Strings.isNullOrEmpty(authRequest.getState())) {
-					uriBuilder.addParameter("state", authRequest.getState()); // copy the state parameter if one was given
-				}
+		Set<SystemScope> sortedScopes = getSortedScopes(authRequest);
+		model.put(SCOPES, sortedScopes);
```

Comment:
Inline the sortedScopes variable.

Reply:
I think it's not a good idea here because the sortedScopes variable is used once more further in the code (so if I inline it, the whole method will execute twice).